### ISSUE TITLE
construct-constructor-property: false positive on props parameter with a default value (props: T = {})

### ISSUE THREAD

**Issue by konokenj** [ren-yamanashi/eslint-cdk-plugin#509]
## Describe the bug

`awscdk/construct-constructor-property` reports a false positive when the third
constructor parameter has a **default value**, e.g. `props: MyConstructProps = {}`.
The parameter is correctly named `props`, but the rule still flags it with
*"Constructor of a class extending Construct must have the property names
'scope, id' or 'scope, id, props'"*.

`props: T` and `props?: T` are both accepted (the latter is explicitly listed as a
Correct example in the docs), so a defaulted `props: T = {}` — which is just as
valid and common — should be accepted too.

## Expected Behavior

No error for a third parameter named `props` that has a default value.

The docs list both of these as ✅ Correct:
- `constructor(scope: Construct, id: string, props: MyConstructProps)`
- `constructor(scope: Construct, id: string, props?: MyConstructProps)`
  (https://awscdk-lint.dev/rules/construct-constructor-property.html)

A defaulted parameter `props: MyConstructProps = {}` uses the same name `props`
and should be treated the same way.

## Current Behavior

The parameter is reported as a naming violation:

```
  × awscdk(construct-constructor-property): Constructor of a class extending Construct must have the property names 'scope, id' or 'scope, id, props'
    ╭─[src/fixture.ts:25:45]
 24 │ export class C extends Construct {
 25 │   constructor(scope: Construct, id: string, props: FixtureProps = {}) {
    ·                                             ────────────────────────
 26 │     super(scope, id);
    ╰────
```

### Minimal reproduction

`src/fixture.ts`:
```ts
import { Construct } from "constructs";

export interface FixtureProps {
  readonly x?: string;
}

// Case A — OK (props: T)
export class A extends Construct {
  constructor(scope: Construct, id: string, props: FixtureProps) {
    super(scope, id);
    void props;
  }
}

// Case B — OK (props?: T, documented as correct)
export class B extends Construct {
  constructor(scope: Construct, id: string, props?: FixtureProps) {
    super(scope, id);
    void props;
  }
}

// Case C — expected OK, but reported (props: T = {})
export class C extends Construct {
  constructor(scope: Construct, id: string, props: FixtureProps = {}) {
    super(scope, id);
    void props;
  }
}
```

`.oxlintrc.json` (single rule enabled):
```json
{
  "jsPlugins": ["oxlint-plugin-awscdk"],
  "rules": {
    "awscdk/construct-constructor-property": "error"
  }
}
```

Command and result:
```
$ oxlint --type-aware --format=unix src/fixture.ts
src/fixture.ts:25:45: Constructor of a class extending Construct must have the property names 'scope, id' or 'scope, id, props' [Error/awscdk(construct-constructor-property)]
```

Only Case C (line 25, `props: FixtureProps = {}`) is reported. Cases A and B are
clean. The same happens without `--type-aware`.

## Possible root cause

`checkThirdParamIsProps` accepts the parameter only when it is an
`AST_NODE_TYPES.Identifier`:

```ts
const checkThirdParamIsProps = (thirdParam) => {
  if (!thirdParam) return;
  if (thirdParam.type !== AST_NODE_TYPES.Identifier || thirdParam.name !== "props") {
    context.report({ node: thirdParam, messageId: "invalidConstructorProperty" });
  }
};
```

A parameter with a default value (`props: T = {}`) is an `AssignmentPattern`
node, not an `Identifier`, so it fails this check. The inner identifier
(`assignmentPattern.left`) is never unwrapped. `props?` stays an `Identifier`
(with `optional: true`), which is why the optional form passes but the defaulted
form does not.

The same logic exists in both packages, so both are likely affected:
- `packages/oxlint/src/rules/construct-constructor-property.ts`
- `packages/eslint/src/rules/construct-constructor-property.ts`

(The first/second parameter checks `checkFirstParamIsScope` / `checkSecondParamIsId`
have the same `!== Identifier` shape, so `scope`/`id` with default values would
likely be affected in the same way, though that is much less common in practice.)

## Environment

- **Plugin**: oxlint-plugin-awscdk
- **Plugin Version**: 0.1.3
- **oxlint**: 1.75.0 (also relevant: plugin peer-requires oxlint `^1.70.0`)
- **Language Version**: TypeScript (via `@typescript/native-preview` 7.x)
- **OS**: macOS 26.5.1
- **Package Manager**: pnpm 10.25.0
- **Node**: v24.11.1

Not specific to a single oxlint version: originally observed on oxlint 1.56.0 and
reproduced here on 1.75.0.

**Comment by ren-yamanashi** [comment]
Thank you for opening the issue!

I have addressed this bug and released the fix in [oxlint-plugin-awscdk@0.1.4](https://www.npmjs.com/package/oxlint-plugin-awscdk)
